File: Patterns/C#/Creational/Builder/BuilderJacketCap.cs
public class BuilderJacketCap : Builder{

    public BuilderJacketCap(){
        outfit = new Outfit();
    }

    public override void buildCoat(){
        outfit.addClothe( new Jacket() );
    }

    public override void buildHat(){
        outfit.addClothe( new Cap() );
    }

    public override Outfit getOutfit(){
        return outfit;
    }

}

File: Patterns/Java/Creational/Builder/BuilderJacketCap.java
public class BuilderJacketCap extends Builder{

    BuilderJacketCap(){
        outfit = new Outfit();
    }

    public void buildCoat(){
        outfit.addClothe( new Jacket() );
    }


    public void buildHat(){
        outfit.addClothe( new Cap() );
    }

    public Outfit getOutfit(){
        return outfit;
    }

}
